src/lib.rs

```diff
@@ -10,16 +10,47 @@
 
 //! Support for matching file paths against Unix shell style patterns.
 //!
-//! The `glob` and `glob_with` functions, in concert with the `Paths`
-//! type, allow querying the filesystem for all files that match a particular
-//! pattern - just like the libc `glob` function (for an example see the `glob`
-//! documentation). The methods on the `Pattern` type provide functionality
-//! for checking if individual paths match a particular pattern, in a similar
-//! manner to the libc `fnmatch` function.
+//! The `glob` and `glob_with` functions allow querying the filesystem for all files
+//! that match a particular pattern (similar to the libc `glob` function). The methods on the
+//! `Pattern` type provide functionality for checking if individual paths match a
+//! particular pattern (similar to the libc `fnmatch` function).
 //!
 //! For consistency across platforms, and for Windows support, this module
 //! is implemented entirely in Rust rather than deferring to the libc
 //! `glob`/`fnmatch` functions.
+//!
+//! # Examples
+//!
+//! To print all jpg files in `/media/` and all of its subdirectories:
+//!
+//! ```rust
+//! use glob::glob;
+//!
+//! for entry in glob("/media/**/*.jpg").unwrap() {
+//!     if let Ok(path) = entry {
+//!         println!("{:?}", path.display())
```

Comment:
Technically no, I think it just uses the value of println to be the value of the if-let block. But this was a typo on my part anyway.